src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/DataPlaneCollectionSetup.cs

```diff
@@ -0,0 +1,159 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Abstractions.Exceptions;
+using Microsoft.Health.Core.Features.Context;
+using Microsoft.Health.Fhir.CosmosDb.Core;
+using Microsoft.Health.Fhir.CosmosDb.Core.Configs;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.Versioning;
+using Polly;
+using Polly.Retry;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Initialization.Features.Storage
+{
+    public class DataPlaneCollectionSetup : ICollectionSetup
+    {
+        private readonly ILogger<DataPlaneCollectionSetup> _logger;
+        private readonly CosmosClient _client;
+        private readonly Lazy<Container> _container;
+        private const int CollectionSettingsVersion = 3;
+        private readonly CosmosDataStoreConfiguration _cosmosDataStoreConfiguration;
+
+        // TODO: move loggerto be the last parameter.
```

Comment:
You can remove the parameter from line 35.

Reply:
done